app/code/Magento/Tax/Model/Calculation/Rule/Validator.php

```diff
@@ -73,7 +73,7 @@ public function isValid($value)
         }
 
         // Code is required
-        if (!\Zend_Validate::is(trim($value->getCode()), 'NotEmpty')) {
+        if (!\Zend_Validate::is(trim($value->getCode() ?? ''), 'NotEmpty')) {
```

Comment:
`if ($taxRate->getCode() === null || !\Zend_Validate::is(trim($taxRate->getCode()), 'NotEmpty')) { ` could be a better option

Reply:
@xmav I changed, thanks for your help!